contracts/SDUtilityPool.sol

```diff
@@ -0,0 +1,906 @@
+
+        /*
+         * Calculate the fee accumulated into utilized and totalProtocolFee and the new index:
+         *  simpleFeeFactor = utilizationRate * blockDelta
+         *  feeAccumulated = simpleFeeFactor * totalUtilizedSD
+         *  totalUtilizedSDNew = feeAccumulated + totalUtilizedSD
+         *  totalProtocolFeeNew = feeAccumulated * protocolFeeFactor + totalProtocolFee
+         *  utilizeIndexNew = simpleFeeFactor * utilizeIndex + utilizeIndex
+         */
+
+        uint256 simpleFeeFactor = utilizationRatePerBlock * blockDelta;
+        uint256 feeAccumulated = (simpleFeeFactor * totalUtilizedSD) / DECIMAL;
+        totalUtilizedSD += feeAccumulated;
+        accumulatedProtocolFee += (protocolFeeFactor * feeAccumulated) / DECIMAL;
+        utilizeIndex += (simpleFeeFactor * utilizeIndex) / DECIMAL;
+
+        accrualBlockNumber = currentBlockNumber;
+
+        emit AccruedFees(feeAccumulated, accumulatedProtocolFee, totalUtilizedSD);
+    }
+
+    /**
+     * @notice Initiates the liquidation process for an account if its health factor is below the required threshold.
+     * @dev The function checks the health factor, accrues fees, updates utilized indices, and calculates liquidation amounts.
+     * @param account The address of the account to be liquidated
+     */
+    function liquidationCall(address account) external override {
+        if (liquidationIndexByOperator[account] != 0) revert AlreadyLiquidated();
+
+        UserData memory userData = getUserData(account);
+
+        if (userData.healthFactor > 1) {
+            revert NotLiquidatable();
+        }
+
+        accrueFee();
+        utilizerData[account].utilizeIndex = utilizeIndex;
+        totalUtilizedSD -= userData.totalInterestSD;
+
+        IERC20(staderConfig.getStaderToken()).transferFrom(msg.sender, address(this), userData.totalInterestSD);
+
+        uint256 sdPriceInEth = IStaderOracle(staderConfig.getStaderOracle()).getSDPriceInETH();
+        uint256 totalInterestInEth = userData.totalInterestSD * sdPriceInEth;
+        uint256 liquidationBonusInEth = (totalInterestInEth * riskConfig.liquidationBonusPercent) / 100;
+        uint256 liquidationFeeInEth = (totalInterestInEth * riskConfig.liquidationFeePercent) / 100;
+        uint256 totalLiquidationAmountInEth = totalInterestInEth + liquidationBonusInEth + liquidationFeeInEth;
+
+        OperatorLiquidation memory liquidation = OperatorLiquidation({
+            totalAmountInEth: totalLiquidationAmountInEth,
+            totalBonusInEth: liquidationBonusInEth,
+            totalFeeInEth: liquidationFeeInEth,
+            isRepaid: false,
+            isClaimed: false,
+            liquidator: msg.sender
+        });
+        liquidations.push(liquidation);
+        liquidationIndexByOperator[account] = liquidations.length;
+
+        IPoolUtils(staderConfig.getPoolUtils()).processOperatorExit(account, totalLiquidationAmountInEth / 4 + 1);
```

Comment:
1. What happens if the nodes are slashed and don't have enough ETH from this count of exited nodes?
2. What if the node count is in 1000s that `processOperatorExit` cannot complete in one call?

Reply:
Changed to `totalAmount` instead.
So the logic moves to offchain 👍